sw/applications/example_power_gating_periph/example_power_gating_periph.c

```diff
@@ -0,0 +1,50 @@
+// Copyright EPFL contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include <stdio.h>
+#include <stdlib.h>
+#include "csr.h"
+#include "hart.h"
+#include "handler.h"
+#include "core_v_mini_mcu.h"
+#include "power_manager.h"
+
+static power_manager_t power_manager;
+
+int main(int argc, char *argv[])
+{
+    // Setup power_manager
+    mmio_region_t power_manager_reg = mmio_region_from_addr(POWER_MANAGER_START_ADDRESS);
+    power_manager.base_addr = power_manager_reg;
+
+    power_manager_counters_t power_manager_periph_counters;
+
+    // Init peripheral_subsystem's counters
+    if (power_gate_counters_init(&power_manager_periph_counters, 40, 40, 30, 30) != kPowerManagerOk_e)
+    {
+        printf("Error: power manager fail. Check the reset and powergate counters value\n");
+        return EXIT_FAILURE;
+    }
+
+    // Power off peripheral_subsystem
+    if (power_gate_domain(&power_manager, kPeriph_e, kOff_e, &power_manager_periph_counters) != kPowerManagerOk_e)
+    {
+        printf("Error: power manager fail.\n");
+        return EXIT_FAILURE;
+    }
+
+    // Wait some time
+    for (int i=0; i<100; i++);
```

Comment:
You could test one of the interrupts here. Add something to check the peripherals are alive again too. You can check they have been properly reset too

Reply:
When I wake-up I use the UART that is quite complex, so it may properly verify the state of the domain, I think. 
I substituted that simple wait with a wait-for-interrupt :)